Fixed #18468 -- Added support for comments on columns and tables. (#14463)
django/django: docs/ref/models/fields.txt

hyphen -- that's OK. Django quotes column and table names behind the
scenes.

``db_comment``
--------------

> .. attribute:: Field.db_comment

Reviewer: The documentation says that new features should have the `versionadded` block ([documentation](https://docs.djangoproject.com/en/3.2/internals/contributing/writing-documentation/#documenting-new-features)) like this:
```
.. versionadded:: 4.0
```
(It looks like the current in-development version is 4.0: [link](https://code.djangoproject.com/wiki/Version4.0Roadmap))
But I'm not 100% clear on whether the author should be adding that block or if that's the job of the release manager (in this case, Mariusz Felisiak). I'll post that question into the django-developers discussion group.

Reviewer: Update: Carlton confirmed that adding the versionadded block is appreciated.

db_comment support SQL COMMENT.

It is similar to `help_text`_ in terms of displaying. it's useful for sharing
documents with DBA as well as developers. If you want to share `help_text`_
with DBA, use ``db_comment`` instead of `help_text`_. For example::

>     db_comment="this is denormalization Field, do not update this field alone."

Reviewer: Suggesting some slight wording changes and the addition of a warning:
```
A string describing the Model field which will be saved as a SQL 
COMMENT on the SQL COLUMN

Unlike `help_text`, db_comments are displayed in the database as
SQL COMMENTs, which makes them useful for documenting fields
for database administrators, developers, analysts, and others with
direct database access, especially if you need to provide different
explanations  for users or site admins (using `help_text`) vs. for DBAs
(using `db_comment`). For example::    

    db_comment="User karma points. This is field is denormalized,\
                          so should not be updated alone. This comment is\
                          managed by Django, and will be updated on every\
                          database migration."

db_comment has a character limit that differs for each database 
backend. PostgreSQL has a 1GB limit. Oracle has a _____ limit. 
MySQL has a 1024 character limit. 

.. warning::
    Changes to database column comments made by database 
    users outside of Django will be overwritten by Django migrations 
    if Django finds a value for db_comment on a model field. 
    Communicating with your DBAs and issuing reminder notices in 
    the comments (as in the example above) may help avoid conflicts.
```

hyphen -- that's OK. Django quotes column and table names behind the
scenes.

``db_comment``
--------------

.. versionadded:: 4.0
.. attribute:: Field.db_comment

> A string describing the Model field which will be saved as a SQL

Author: According to the [Oracle docs](http://docs.oracle.com/database/121/CNCPT/sqllangu.htm#CNCPT116) it is sequel, so “a SQL …” is the correct way

refer: https://oracle-base.com/blog/2015/01/02/a-sql-or-an-sql/

> A string describing the Model field which will be saved as a SQL

Reviewer 2: Guidelines from our docs are more important, see [Commonly used terms](https://docs.djangoproject.com/en/4.1/internals/contributing/writing-documentation/#commonly-used-terms).

COMMENT on the SQL COLUMN

Unlike ``help_text``, db_comments are displayed in the database as
SQL comments, which is useful for documenting fields
for database administrators, developers, analysts, and others with
direct database access, especially if you need to provide different

> explanations  for users or site admins (using `help_text`) vs. for DBAs

Author: Thank you for your feedback on my terrible writing skill 😄

hyphen -- that's OK. Django quotes column and table names behind the
scenes.

> ``db_comment``

Reviewer 3: I think this whole section should be moved underneath the others, I think the others are more important and things should generally be added at the bottom.

Reviewer: The list is alphabetized. This should remain in its current location so that the list remains alphabetized.

--------------

.. versionadded:: 4.0
.. attribute:: Field.db_comment

> db_comment support SQL COMMENT.

Reviewer 3: I think we can remove this whole line.

A string describing the Model field which will be saved as a SQL
COMMENT on the SQL COLUMN

Unlike `help_text`, db_comments are displayed in the database as

> SQL COMMENTs, which makes them useful for documenting fields

Reviewer 3: ```suggestion
SQL comments, which is useful for documenting fields
```

for database administrators, developers, analysts, and others with
direct database access, especially if you need to provide different
explanations  for users or site admins (using `help_text`) vs. for DBAs
(using `db_comment`). For example::

    db_comment = "User karma points. This is field is denormalized,\
                          so should not be updated alone. This comment is\
                          managed by Django, and will be updated on every\
                          database migration."

db_comment has a character limit that differs for each database
backend. PostgreSQL has a 1GB limit. MySQL has a 1024 character limit.

.. warning::
    Changes to database column comments made by database
    users outside of Django will be overwritten by Django migrations
    if Django finds a value for db_comment on a model field.
    Communicating with your DBAs and issuing reminder notices in

>     the comments (as in the example above) may help avoid conflicts.

Reviewer 3: I think I mentioned this before, but as it applies to more or less everything in Django, I think we don't need it.

Reviewer: I agree. This can be removed.

hyphen -- that's OK. Django quotes column and table names behind the
scenes.

``db_comment``
--------------

.. versionadded:: 4.2
.. attribute:: Field.db_comment

A string describing the Model field which will be saved as an SQL
COMMENT on the SQL COLUMN

Unlike ``help_text``, db_comments are displayed in the database as
SQL comments, which is useful for documenting fields

> for database administrators, developers, analysts, and others with

Reviewer 4: ```suggestion
for individuals with
```

direct database access, especially if you need to provide different
explanations for users or site admins (using ``help_text``) compared to DBAs
(using ``db_comment``). For example::

    db_comment = "User karma points. This is field is denormalized,\
                          so should not be updated alone. This comment is\
                          managed by Django, and will be updated on every\
                          database migration."

``db_comment`` has a character limit that differs for each database

> backend. PostgreSQL has a 1GB limit. MySQL has a 1024 character limit.

Reviewer 4: I wonder if this part. Ould be generalised?

As written I'm asking what about oracle / sqlite? Do these limits change between versions too? That could be all too much for the docs.
